spec/Entity/InvoiceSpec.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace spec\Sylius\InvoicingPlugin\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\InvoicingPlugin\Entity\BillingDataInterface;
+use Sylius\InvoicingPlugin\Entity\InvoiceInterface;
+use Sylius\InvoicingPlugin\Entity\LineItemInterface;
+
+final class InvoiceSpec extends ObjectBehavior
+{
+    function let(
+        BillingDataInterface $billingData,
+        LineItemInterface $lineItem
+    ): void {
+        $issuedAt = new \DateTimeImmutable('now');
+
+        $this->beConstructedWith(
+            '7903c83a-4c5e-4bcf-81d8-9dc304c6a353',
+            '007',
+            $issuedAt,
+            $billingData,
+            'USD',
+            300,
+            10300,
+            new ArrayCollection([$lineItem->getWrappedObject()])
+        );
+    }
+
+    function it_implements_invoice_interface(): void
+    {
+        $this->shouldImplement(InvoiceInterface::class);
+    }
+
+    function it_implements_resource_interface(): void
+    {
+        $this->shouldImplement(ResourceInterface::class);
+    }
+
+    function it_has_an_id(
+        BillingDataInterface $billingData,
+        LineItemInterface $lineItem
+    ): void {
+        $issuedAt = new \DateTimeImmutable('now');
+        $lineItems = new ArrayCollection([$lineItem->getWrappedObject()]);
+
+        $this->beConstructedWith(
```

Comment:
Since you don't need to change constructor parameters in several tests, why don't you put this in let() function?

Reply:
In this case, I need this function here to properly check not primitive parameters